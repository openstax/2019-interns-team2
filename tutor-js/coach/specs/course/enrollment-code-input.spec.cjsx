{React, Testing, sinon, _, ReactTestUtils} = require 'shared/specs/helpers'
SnapShot = require 'react-test-renderer'

jest.mock('navigation/model')
Navigation = require 'navigation/model'

EnrollmentCodeInput = require 'course/enrollment-code-input'
Course = require 'course/model'
STATUS = require '../../auth/status/GET'

describe 'EnrollmentCodeInput Component', ->

  beforeEach ->
    @props =
      title: 'Join my Course'
      isTeacher: false
      course: new Course(ecosystem_book_uuid: 'test-uuid')
      currentCourses: _.map STATUS.courses, (c) -> new Course(c)

  it 'displays second semester message when prop is set', ->
    @props.secondSemester = true
    wrapper = shallow(<EnrollmentCodeInput {...@props} />)
    expect(wrapper.text()).to.include('A New Semester, A New Enrollment Code')
    expect(wrapper.find('.past-course')).to.have.length(1)
    expect(SnapShot.create(<EnrollmentCodeInput {...@props} />).toJSON()).toMatchSnapshot()

    undefined

  it 'model registers when submit is clicked', ->
    sinon.stub(@props.course, 'register')
    Testing.renderComponent( EnrollmentCodeInput, props: @props ).then ({dom}) =>
      dom.querySelector('input').value = 'test'
      Testing.actions.click(dom.querySelector('.btn-default'))
      expect(@props.course.register).to.have.been.calledWith('test')

  it 'model registers when enter is pressed in input', ->
    sinon.stub(@props.course, 'register')
    Testing.renderComponent( EnrollmentCodeInput, props: @props ).then ({dom}) =>
      input = dom.querySelector('input')
      input.value = 'test'
      ReactTestUtils.Simulate.keyPress(input, {key: "Enter"})
      expect(@props.course.register).to.have.been.calledWith('test')
