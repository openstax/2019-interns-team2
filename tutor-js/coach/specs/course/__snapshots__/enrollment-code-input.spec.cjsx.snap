// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollmentCodeInput Component displays second semester message when prop is set 1`] = `
<div
  className="enrollment-code form-group row"
>
  <div
    className="second-semester"
  >
    <h3
      className="text-center"
    >
      A New Semester, A New Enrollment Code
    </h3>
    <p>
      Concept Coach requires a new enrollment code each semester.
      If youâ€™re in the second semester of a two-semester course,
      get the new code from your instructor and enter it below.
    </p>
  </div>
  <hr />
  <div
    className="code-wrapper col-md-6 col-md-offset-3 col-sm-8 col-sm-offset-2 col-xs-12"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="enrollment-code"
      >
        Enter the enrollment code
      </label>
      <span
        className="input-group"
      >
        <input
          autoFocus={true}
          className="form-control"
          id="enrollment-code"
          onKeyPress={[Function]}
          placeholder="enrollment code"
          type="text"
        />
        <span
          className="input-group-btn"
        >
          <button
            className="async-button enroll btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Enroll
          </button>
        </span>
      </span>
    </div>
    <button
      className="past-course btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Return to my past Concept Coach course
    </button>
  </div>
</div>
`;
